src/app/_components/DistrictSearch/actions.js

```diff
@@ -0,0 +1,11 @@
+'use server';
+
+import { prisma } from '@/lib/db';
+
+export const getDistrictsList = async () => {
+  const districtsList = await prisma.district.findMany({
+    orderBy: { name: 'desc' },
+  });
```

Comment:
1. Is it possible that you may have large amount of districts? If so, does it make sense to limit search?
2. It seems you are relying on sorting districts on the backend to display sorted list on the frontend. Would it make sense to make sure it's sorted properly on the frontend instead?

Reply:
just out of curiosity - why filtering on frontend rather than on backend?

Reply:
Again loose-coupling: order seems to be a frontend-feature, not backend.
Unless you implement pagination or some other order-related logic, of course.

Tight coupling leads to errors. E.g. at some point you decide to store response in frontend-storage, or cache it.
And most likely at that point you won't think about necessity to change smth about ordering in components that use it.

Reply:
1. It's all about Lviv districts, so I'd say that the amount will stay the same for quite a long time. However, in case the amount were bigger, what would be the way to limit search? Pagination? Because all the districts should be available for user to pick from.

2. I though it would be a good idea to have server do all the work including sorting out the districts. What are the benefits of doing it on the front end?

Reply:
1. OK, we can leave it this way for now. If we have them too many, maybe we'd need to implement smth like dropdown search that makes requests based on input.
2. Just answered above. That's also about loose coupling: it would definitely work but you might run into issues long-term if you follow this approach.

Reply:
Got it. Thanks